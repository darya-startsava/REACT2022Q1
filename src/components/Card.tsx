import CardType from '../types/card';

export default function Card({ id, image, name, year, director, actors, imdb, oscars }: CardType) {
  return (
    <div>
      <img src={`posters/${image}`} width="200" height="auto" alt="poster" />
      <h5>&ldquo;{name}&rdquo;</h5>
      <p>{year} year</p>
      <p>Director: {director}</p>
      <p>Starring: {actors}</p>
      <p>IMDb: {imdb}/10</p>
      <p>Oscars: {oscars}</p>
    </div>
  );
}
