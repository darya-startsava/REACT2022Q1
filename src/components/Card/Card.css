.card-image {
  width: 200px;
  height: auto;
}

.card-item {
  width: 200px;
  height: auto;
}

.close-img {
  width: 20px;
  height: 20px;
}

.close-button {
  border: none;
  background-color: #fff;
  margin-left: 180px;
}