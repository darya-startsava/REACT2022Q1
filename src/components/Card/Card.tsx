import './Card.css';
import CardType from '../../types/card';

export default function Card({
  id,
  image,
  uploadedImage,
  name,
  year,
  director,
  actors,
  imdb,
  oscars,
  gender,
  dateOfBirth,
  countryOfBirth,
  movieGenres,
  overview,
  releaseDate,
  voteAverage,
  isFull,
}: CardType) {
  let style = {};
  if (isFull) {
    style = { width: 500 };
  }
  return (
    <div className="my-1 d-flex flex-column card-item" style={style} data-testid="card">
      {image && <img src={image} className="align-self-center card-image" alt="poster" />}
      {uploadedImage && <img src={uploadedImage} className="card-image" alt="avatar" />}
      <h5 className="align-self-center card-title">{name}</h5>
      {year && <p>{year} year</p>}
      {director && <p>Director: {director}</p>}
      {actors && <p>Starring: {actors ? actors.join(', ') : ''}</p>}
      {imdb && <p>IMDb: {imdb}/10</p>}
      {oscars && <p>Oscars: {oscars}</p>}
      {gender && <p>{gender}</p>}
      {dateOfBirth && <p>Date of birth: {dateOfBirth}</p>}
      {countryOfBirth && <p>Country of birth: {countryOfBirth}</p>}
      {movieGenres && <p>Genres: {movieGenres ? movieGenres.join(', ') : ''}</p>}
      {isFull && <p className="overview px-2">Overview: {overview}</p>}
      {isFull && <p className="px-2">Release date: {releaseDate}</p>}
      {isFull && <p className="px-2">Vote average: {voteAverage}</p>}
    </div>
  );
}
