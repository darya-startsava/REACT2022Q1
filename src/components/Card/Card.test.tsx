import { render } from '@testing-library/react';

import Card from './Card';

describe('Card', () => {
  test('should contain limited information about film if it is not modal', () => {
    const { container } = render(
      <Card
        id={0}
        name="title0"
        image="no-image.jpg"
        overview="overview0"
        releaseDate="release_date0"
        voteAverage="0"
        isFull={false}
      />
    );
    expect(container).toMatchSnapshot();
  });

  test('should contain full information about film if it is modal', () => {
    const { container } = render(
      <Card
        id={0}
        name="title0"
        image="no-image.jpg"
        overview="overview0"
        releaseDate="release_date0"
        voteAverage="0"
        isFull={true}
      />
    );
    expect(container).toMatchSnapshot();
  });
});
