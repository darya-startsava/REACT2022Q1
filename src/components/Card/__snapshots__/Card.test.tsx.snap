// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should contain full information about film if it is modal 1`] = `
<div>
  <div
    class="my-1 d-flex flex-column card-item"
    data-testid="card"
    style="width: 500px;"
  >
    <img
      alt="poster"
      class="align-self-center card-image"
      src="no-image.jpg"
    />
    <h5
      class="align-self-center card-title"
    >
      title0
    </h5>
    <p
      class="overview px-2"
    >
      Overview: 
      overview0
    </p>
    <p
      class="px-2"
    >
      Release date: 
      release_date0
    </p>
    <p
      class="px-2"
    >
      Vote average: 
      0
    </p>
  </div>
</div>
`;

exports[`Card should contain limited information about film if it is not modal 1`] = `
<div>
  <div
    class="my-1 d-flex flex-column card-item"
    data-testid="card"
    style="cursor: pointer;"
  >
    <img
      alt="poster"
      class="align-self-center card-image"
      src="no-image.jpg"
    />
    <h5
      class="align-self-center card-title"
    >
      title0
    </h5>
  </div>
</div>
`;
